source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift

```diff
@@ -0,0 +1,96 @@
+import AdaptiveCards_bridge
+import AppKit
+
+class FactSetRenderer: NSObject, BaseCardElementRendererProtocol {
+    static let shared = FactSetRenderer()
+    
+    func render(element: ACSBaseCardElement, with hostConfig: ACSHostConfig, style: ACSContainerStyle, rootView: NSView, parentView: NSView, inputs: [BaseInputHandler]) -> NSView {
+        guard let factSet = element as? ACSFactSet else {
+            logError("Element is not of type ACSFactSet")
+            return NSView()
+        }
+        let factArray = factSet.getFacts()
+        let factsetConfig = hostConfig.getFactSet()
+        
+        // Create horizontal Stack to add both views
+        let horizontalStack = ACRFactSetStackView()
+        horizontalStack.orientation = NSUserInterfaceLayoutOrientation.horizontal
+        horizontalStack.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Create stacks to hold title and value of FactSet
+        let titleStack = NSStackView()
+        let valueStack = NSStackView()
+        titleStack.orientation = .vertical
+        valueStack.orientation = .vertical
+        titleStack.translatesAutoresizingMaskIntoConstraints = false
+        valueStack.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Main loop to iterate over Array of facts
+        for fact in factArray {
+            let titleView = TitleFact()
+            let valueView = ACRFactTextField()
+            titleView.titleText.setLabel(string: fact.getTitle())
+            valueView.setLabel(string: fact.getValue())
+            
+            if let colorHex = hostConfig.getForegroundColor(style, color: .default, isSubtle: false), let textColor = ColorUtils.color(from: colorHex) {
+                titleView.titleText.labelText.textColor = textColor
+                valueView.labelText.textColor = textColor
+            }
+            
+            if !(titleView.titleText.labelText.stringValue.isEmpty) || !(valueView.labelText.stringValue.isEmpty) {
+                titleStack.addArrangedSubview(titleView.titleText)
+                valueStack.addArrangedSubview(valueView)
+            }
+        }
+        
+        // Add both elements into the Horizontal Stack
+        horizontalStack.addArrangedSubview(titleStack)
+        horizontalStack.addArrangedSubview(valueStack)
+        
+        // Add constraints once all the views are in the same view heirarchy
+        let horizontalStackViews = horizontalStack.arrangedSubviews
+        guard let titleStackagain = horizontalStackViews[0] as? NSStackView else { return NSView() }
+        guard let valueStackagain = horizontalStackViews[1] as? NSStackView else { return NSView() }
+        
```

Comment:
Why do we need these lines? Can't we directly use `titleStack` and `valueStack` here? I just see it added into the stack in the previous line.

Reply:
I had to do this to make them in the same view heirarchy, so that I can constraint them against each other and set height and stuff.
So I need this basically

Reply:
Ahh that's fine. It's added in the stack in the previous line. Do we need these downcasting here?

Reply:
Yeah akaash
Otherwise it returns it as NSview and then can't access stackview properties needed to set height

![image](https://user-images.githubusercontent.com/78855675/109269426-7a456c00-7832-11eb-9f02-a03aea6ceabd.png)

Reply:
`titleStack` can be directly used. Will sync offline on this.

Reply:
Ok cool cool
Thanks Akaash!